### DISCUSSION TITLE
Suggerstion: Adding Psl\Iter\chunk

### DISCUSSION THREAD

**Discussion by derflocki** [azjezz/psl#421] [General]
I was wondering if you would mind the addition of `Psl\Iter\chunk`? Since there is `CollectionInterface::chunk` I was surprised that it's not implemented.

Would be a great way for me to contribute a PR.

**Comment by veewee** [comment]
PSL split up iterable functions into:

* `Psl\Vec` : Operates on iterables and returns a vec
* `Psl\Dict` : Operates on iterables and returns a dict
* `Psl\Iter` : Operates on iterables but returns nothing or something mixed

You are probably looking for https://github.com/azjezz/psl/blob/next/src/Psl/Vec/chunk.php ?

**Reply by derflocki** [reply]
Thanks for the clarification. I was looking for something that yields chunks on the fly. `Vec\chunk` "materializes" the iterable, so it's not working for my current use-case.
